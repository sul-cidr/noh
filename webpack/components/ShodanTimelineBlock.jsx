import React from "react";
import PropTypes from "prop-types";

// Rather than passing in style by props, these should be
// calculated from duration of section and intensity
const ShodanTimelineBlock = props => {
  // should actually check to be sure that intensity is not greater than max
  const heightNum = parseInt(props.intensity, 10) / props.maxIntensity * 100;
  const durationNum = props.duration / props.totalDuration * 100;
  return (
    <div
      className={`shodan-map__item ${props.url ? "pointer" : ""}`}
      style={{
        left: props.left,
        width: `${durationNum}%`,
        height: `${heightNum}%`
      }}
      data-tooltip={props.name}
      onClick={() => props.url && window.location.assign(props.url)}
      role="link"
      tabIndex={0}
      onKeyPress={null}
    />
  );
};

ShodanTimelineBlock.propTypes = {
  left: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  url: PropTypes.string,
  maxIntensity: PropTypes.number.isRequired,
  intensity: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
  totalDuration: PropTypes.number.isRequired
};

ShodanTimelineBlock.defaultProps = {
  url: ""
};

export default ShodanTimelineBlock;
