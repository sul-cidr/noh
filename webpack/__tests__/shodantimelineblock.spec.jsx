import React from "react";
import { mount, shallow } from "enzyme";
import ShodanTimelineBlock from "../components/ShodanTimelineBlock";

describe("<ShodanTimelineBlock>", () => {
  it("renders as expected", () => {
    const component = shallow(
      <ShodanTimelineBlock
        name="Kiri"
        left="8%"
        intensity="15"
        maxIntensity={21}
        duration={10}
        totalDuration={30}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("renders as expected with url", () => {
    const component = shallow(
      <ShodanTimelineBlock
        name="Kiri"
        url="/kokaji/kiri"
        left="8%"
        intensity="15"
        maxIntensity={21}
        duration={10}
        totalDuration={30}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("navigates if clicked and url", () => {
    window.location.assign = jest.fn();
    const url = "/kokaji/kiri";
    const wrapper = mount(
      <ShodanTimelineBlock
        name="Kiri"
        url={url}
        left="8%"
        intensity="15"
        maxIntensity={21}
        duration={10}
        totalDuration={30}
      />
    );
    wrapper.find("div").simulate("click");
    expect(window.location.assign).toBeCalledWith(url);
  });

  it("does not navigate if clicked and no url", () => {
    window.location.assign = jest.fn();
    const wrapper = mount(
      <ShodanTimelineBlock
        name="Kiri"
        url=""
        left="8%"
        intensity="15"
        maxIntensity={21}
        duration={10}
        totalDuration={30}
      />
    );
    wrapper.find("div").simulate("click");
    expect(window.location.assign).not.toBeCalled();
  });
});
