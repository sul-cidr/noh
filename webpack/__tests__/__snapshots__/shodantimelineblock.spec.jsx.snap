// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShodanTimelineBlock> renders as expected 1`] = `
<div
  className="shodan-map__item "
  data-tooltip="Kiri"
  onClick={[Function]}
  onKeyPress={null}
  role="link"
  style={
    Object {
      "height": "71.42857142857143%",
      "left": "8%",
      "width": "33.33333333333333%",
    }
  }
  tabIndex={0}
/>
`;

exports[`<ShodanTimelineBlock> renders as expected with url 1`] = `
<div
  className="shodan-map__item pointer"
  data-tooltip="Kiri"
  onClick={[Function]}
  onKeyPress={null}
  role="link"
  style={
    Object {
      "height": "71.42857142857143%",
      "left": "8%",
      "width": "33.33333333333333%",
    }
  }
  tabIndex={0}
/>
`;
